tests/extractor/extract/test_broadcast_events.py

```diff
@@ -0,0 +1,48 @@
+import unittest
+from unittest import mock
+
+from common.service.errors import BatchCancelled
+from extractor.errors import EmptyResult
+from extractor.service import broadcast_events
+
+
+class BroadCastEventsTestCase(unittest.TestCase):
+    def test_skipping_redis_call_when_batch_cancelled(self):
+        extractor = mock.MagicMock()
+        extractor.split_dataframe.return_value = [1, 2, 3]
+        producer = mock.MagicMock()
+        producer.produce_event.side_effect = BatchCancelled
```

Comment:
```suggestion
        producer.produce_event.side_effect = [None, BatchCancelled]
        ...
        assert producer.produce_event.call_count == 2
        redis.hset.assert_not_called()
```
don't know if it's a lot more useful but it's more faithful to the actual behaviour (cancelling the batch after it started)